### ISSUE TITLE
Query database more times than expected during replay

### ISSUE THREAD

**Issue by yanli1993** [arextest/arex-agent-java#222]
**Bug description**
Query database more times than expected during replay, for one time when record but three times when replay.

**Steps to reproduce**
code：
```
    @Override
    public String testSubInvokeParams(){
        long id = 2L;
        return getUserById(id).getName()+ "mock";
    }

   @Override
    public User getUserById(Long id){
        User user = new User();
        user.setId(id);
        return userMapper.selectOne(user);
    }

tk.mybatis.mapper.common.base.select.SelectOneMapper#selectOne
```

record：
<img width="1022" alt="image" src="https://github.com/arextest/arex-agent-java/assets/47803307/315c7ce4-6a60-4366-a0c6-236aba039466">

replay：
<img width="1046" alt="image" src="https://github.com/arextest/arex-agent-java/assets/47803307/d225961c-f5ed-4ea2-974c-d5f712f4564d">
<img width="1112" alt="image" src="https://github.com/arextest/arex-agent-java/assets/47803307/5e62b731-33f8-490f-b587-b3ceeebd571d">

log:
```
[http-nio-8076-exec-1] INFO io.arex.inst.runtime.log.LogManager - [[title=arex.replay.DynamicClass]]arex replay operation: java.lang.System.currentTimeMillis
[http-nio-8076-exec-1] INFO io.arex.inst.runtime.log.LogManager - [[title=arex.enter]]arex-record-id: AREX-192-168-0-109-4931023370, arex-replay-id: AREX-192-168-0-109-4945562290
[http-nio-8076-exec-1] INFO io.arex.agent.instrumentation.TransformListener - [arex] onTransformation: org.apache.ibatis.executor.BaseExecutor loaded: false from classLoader sun.misc.Launcher$AppClassLoader@18b4aac2
[http-nio-8076-exec-1] INFO io.arex.inst.runtime.log.LogManager - [[title=arex.replay.Database]]arex replay operation: query
[http-nio-8076-exec-1] INFO io.arex.inst.runtime.log.LogManager - [[title=arex.replay.Servlet]]arex replay operation: /testSubInvokeParams4
[http-nio-8076-exec-2] INFO io.arex.inst.runtime.log.LogManager - [[title=arex.replay.DynamicClass]]arex replay operation: java.lang.System.currentTimeMillis
[http-nio-8076-exec-2] INFO io.arex.inst.runtime.log.LogManager - [[title=arex.enter]]arex-record-id: AREX-192-168-0-109-4931023370, arex-replay-id: AREX-192-168-0-109-4945576581
[http-nio-8076-exec-2] INFO io.arex.inst.runtime.log.LogManager - [[title=arex.replay.Database]]arex replay operation: query
[http-nio-8076-exec-2] INFO io.arex.inst.runtime.log.LogManager - [[title=arex.replay.Servlet]]arex replay operation: /testSubInvokeParams4
[http-nio-8076-exec-3] INFO io.arex.inst.runtime.log.LogManager - [[title=arex.replay.DynamicClass]]arex replay operation: java.lang.System.currentTimeMillis
[http-nio-8076-exec-3] INFO io.arex.inst.runtime.log.LogManager - [[title=arex.enter]]arex-record-id: AREX-192-168-0-109-4931023370, arex-replay-id: AREX-192-168-0-109-4945579582
[http-nio-8076-exec-3] INFO io.arex.inst.runtime.log.LogManager - [[title=arex.replay.Database]]arex replay operation: query
[http-nio-8076-exec-3] INFO io.arex.inst.runtime.log.LogManager - [[title=arex.replay.Servlet]]arex replay operation: /testSubInvokeParams4

```

**Expected behavior**
query db once when replay

**What went wrong?**
query db 3 times when replay (with then same params)

**Environment**

Compiler:  (AdoptOpenJDK)(build 1.8.0_275-b01)
arex-agent-java: (e.g., "0.3.7")

**Additional context**

**Comment by YongwuHe** [comment]
This issue has been fixed, please upgrade to version 0.4.2.

**Comment by QizhengMo** [comment]
Please refer to http://www.arextest.com/docs/chapter1/Quick%20Installation/#update for version updating guide.
